### PR TITLE
Private APIs on separate port

### PR DESCRIPTION
**Pull request by benjamin-hergert-hs**
- Split up APIs into public and private APIs based on the port that they are running on
- Moved dependencies in Main.kt into fields that can be mocked under test (might want to consider Guice for DI?)
- Added tests for private/public API flow

### CODE REVIEW

File: backend/src/main/kotlin/com/hootsuite/hermes/Main.kt [hunk 1]
```diff
@@ -34,22 +34,47 @@ import io.ktor.request.header
 import io.ktor.request.receive
 import io.ktor.response.respond
 import io.ktor.response.respondText
-import io.ktor.routing.get
-import io.ktor.routing.post
-import io.ktor.routing.routing
+import io.ktor.routing.*
```
Comment (andy-fisher-hs) [inline backend/src/main/kotlin/com/hootsuite/hermes/Main.kt]: ⛔️ No wildcards.

File: backend/src/main/kotlin/com/hootsuite/hermes/Main.kt [hunk 2]
```diff
@@ -73,29 +97,38 @@ fun main(args: Array<String>) {
...
 
             // Install Slack App
-            get(Config.Endpoint.INSTALL) { installGet(call) }
+            get(Config.Endpoint.INSTALL) { if (ensurePrivateApi(call)) { installGet(call) } }
 
             // Handle Slack Slash Command
-            post(Config.Endpoint.SLACK) { slackPost(call) }
+            post(Config.Endpoint.SLACK) { if (ensurePrivateApi(call)) { slackPost(call) } }
         }
+
```
Comment (andy-fisher-hs) [inline backend/src/main/kotlin/com/hootsuite/hermes/Main.kt]: ⛔️ Redundant lines.

File: backend/src/main/kotlin/com/hootsuite/hermes/github/GithubEventHandler.kt [hunk 3]
```diff
@@ -19,11 +19,9 @@ import com.hootsuite.hermes.model.ReviewState
 import com.hootsuite.hermes.slack.SlackMessageHandler
 
 /**
- * Object to handle events from Github webhooks
+ * Class to handle events from Github webhooks
  */
-object GithubEventHandler {
-
-    private const val GITHUB_MENTION = "@"
+class GithubEventHandler(val dataStore: DataStore) {
```
Comment (andy-fisher-hs) [inline backend/src/main/kotlin/com/hootsuite/hermes/github/GithubEventHandler.kt]: ⛔️ Make this a private val?

File: backend/src/main/kotlin/com/hootsuite/hermes/slack/SlashCommandHandler.kt [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+import com.hootsuite.hermes.database.DataStore
 import com.hootsuite.hermes.model.User
 import com.hootsuite.hermes.slack.model.SlashCommand
 
 /**
- * Object to parse the incoming slash commands from slack
+ * Class to parse the incoming slash commands from slack
  */
-object SlashCommandHandler {
+class SlashCommandHandler(val dataStore: DataStore) {
```
Comment (andy-fisher-hs) [inline backend/src/main/kotlin/com/hootsuite/hermes/slack/SlashCommandHandler.kt]: ⛔️ private

File: backend/src/main/kotlin/com/hootsuite/hermes/Main.kt (hunk 1 above)
Comment (benjamin-hergert-hs) [reply backend/src/main/kotlin/com/hootsuite/hermes/Main.kt]: Why IntelliJ has this set as a default is beyond me...

**Pull request merged**